tests.js

```diff
@@ -1,44 +1,208 @@
 require('dotenv').load();
+
+var _ = require('underscore');
+var chai = require('chai');
+var firebase = require('firebase');
+var fs = require('fs');
 var FreshBooks = require('./freshbooks.js');
-var assert = require('chai').assert;
+var request = require('request');
+
+
 
-describe('FreshBooks Client', function() {
-  var freshbooks;
-  before(function() {
-    var access_token = process.env.FBOOKS_ACCESS_TOKEN;
-    var refresh_token = process.env.FBOOKS_REFRESH_TOKEN;
-    if (!access_token || !refresh_token)
-      throw new Error('Required environment variables FBOOKS_ACCESS_TOKEN and/or FBOOKS_REFRESH_TOKEN were not populated');
+describe('freshbooks', function() {
+  this.timeout(1000 * 20);
+
+  var token;
+  var freshbooks, db;
+  var biz_id, project_id, time_entry_id;
+
+  function initFirebase(callback) {
+    var config = {
+      apiKey: process.env.FIREBASE_API_KEY,
+      databaseURL: process.env.FIREBASE_DB_URL
+    };
+    firebase.initializeApp(config);
+    firebase.auth().signInWithEmailAndPassword(
+      process.env.FIREBASE_USERNAME,
+      process.env.FIREBASE_PASSWORD
```

Comment:
@schuttsm: this make me nervous; can we switch to an auth token? This should probably be a new free-tier Firebase account (if it's not already).

Reply:
@prust Yes, this is a separate, free-tier firebase account.  The only data in it is a single "env_data" object with either a "code" or "refresh_token" property.  If the account becomes compromised, all they can do is screw up the refresh token so the build fails.

I'm open to switching to an auth token (seems a bit overkill, but better safe than sorry when it comes to security).  Did you want one generated with firebase-token-generator like in daycast?  It looks like that method is deprecated in the version of firebase we're using.

The new way is to generate a private key & connect with an admin account and then generate a token that way.  Again, I don't mind doing that, but storing a private key seems the same as storing a password.  What do you think?

Reply:
@schuttsm: I'm not familiar with the new Firebase (and I'm hoping to transition Daycast directly from the old one to Postgres, so I'm not super-interested in learning), so do what seems right to you. I don't want to fight the framework -- signing into a service w/ creds smelled like an antipattern to me, but if that's the typical/recommended way w/ the new Firebase, so be it.